src/ai/backend/manager/api/admin.py

```diff
@@ -110,8 +117,11 @@ async def _handle_gql_common(request: web.Request, params: Any) -> ExecutionResu
     })
 )
 async def handle_gql(request: web.Request, params: Any) -> web.Response:
-    result = await _handle_gql_common(request, params)
-    return web.json_response(result.formatted, status=200)
+    try:
+        result = await _handle_gql_common(request, params)
+        return web.json_response(result.formatted, status=200)
+    except asyncio.TimeoutError:
+        return web.json_response(status=504)
```

Comment:
We should have a structured error output following `ai.backend.manager.api.exceptions` module.